modules/ROOT/pages/userlookup.adoc

```diff
@@ -0,0 +1,191 @@
+The {productname} User Lookup API enables integrations to retrieve and cache user details (such as names and avatars) and identify the current user within the editor.
+
+This is useful when building features that rely on user context, such as commenting, or displaying a list of elements container user information, and lets you manage user data efficiently with built-in caching.
+
+== Using the User Lookup API
+
+Follow the steps below to configure and use the `userLookup` API.
+
+[[step-1-set-the-current-user]]
+=== 1. Set the current user_id
+
+The first step is to tell the editor which user is currently active. This is done using the `+user_id+` initialization option.
+
+include::partial$configuration/user_id.adoc[leveloffset=+2]
+
+To access it:
+
+[source,js]
+----
+editor.on('init', () => {
+  console.log('Current user:', editor.userLookup.userId);
+});
+----
+
+[[step-2-provide-a-user-fetcher]]
+=== 2. Provide a function to fetch user details
+
+To simulate a backend lookup, define a local object that maps user IDs to their corresponding data. 
+
+include::partial$configuration/fetch_users.adoc[leveloffset=+2]
+
+.Example
+[source,js]
+----
+const userDb = {
+  'janedoe': { 
+    id: 'janedoe', 
+    name: 'Jane Doe', 
+    avatar: 'https://example.com/avatar/jane.png' 
+  },
+  'johnsmith': { 
+    id: 'johnsmith', 
+    name: 'John Smith' 
+  },
+  'alextaylor': { 
+    id: 'alextaylor', 
+    name: 'Alex Taylor', 
+    avatar: 'https://example.com/avatar/alex.png' 
+  }
+};
+
+tinymce.init({
+  selector: '#editor',
+  user_id: 'alextaylor',
+  fetch_users: (userIds) => {
+    return Promise.all(
+      userIds.map(
+        (userId) => new Promise(
+          (resolve) => resolve(userDb[userId] || { id: userId })
+        )
+      )
+    )
+  },
+});
+----
+
+This example demonstrates how to simulate user lookup in-browser. In a production environment, this function could be swapped out for a network request.
+
+The returned array should include objects with:
+
+* `id` (string) - Required
+* `name` (string) - Optional (defaults to `id`)
+* `avatar` (string, URL or data URI) - Optional (auto-generated SVG fallback if missing)
+* `custom` (object) - Optional metadata
+
+[[step-3-fetch-user-information]]
+=== 3. Fetch user information
+
+Once the editor is initialized, use the API to look up multiple users:
+
+.Example
+[source,js]
+----
+editor.on('init', () => {
+  const users = editor.userLookup.fetchUsers(['user-1', 'user-2']);
+
+  users['user-1'].then((user) => {
+    console.log('User 1 name:', user.name);
+  });
+
+  Promise.all(Object.values(users)).then((allUsers) => {
+    allUsers.forEach((user) => {
+      console.log(`Fetched: ${user.name} (${user.id})`);
+    });
+  });
+});
+----
+
+NOTE: All lookups are cached. If a user has already been fetched, the same promise will be returned without calling `fetch_users` again.
+
+[[step-4-handle-missing-or-default-users]]
+=== 4. Handle default users if `fetch_users` is not provided
```

Comment:
This is not actually a step. It's what happens if step 2 is not completed. But if it remains a step, it should at least be clear about what it contains.
```suggestion
=== 4. Defaults users if `fetch_users` is not provided
```

Reply:
This has been resolved in recent update.